freqtrade/exchange/exchange.py

```diff
@@ -877,42 +877,67 @@ def get_dry_market_fill_price(self, pair: str, side: str, amount: float, rate: f
 
         return rate
 
-    def _is_dry_limit_order_filled(self, pair: str, side: str, limit: float) -> bool:
+    def _fill_dry_limit_order(self, pair: str, side: str,
+                              limit: float, given_amount: float) -> Tuple[Optional[float], float]:
+        """
+        Returns average price and filled amount
+        """
         if not self.exchange_has('fetchL2OrderBook'):
-            return True
-        ob = self.fetch_l2_order_book(pair, 1)
-        try:
+            return limit, given_amount
+        for order_book_top in (None, 1000):
+            ob = self.fetch_l2_order_book(pair, order_book_top)
```

Comment:
Passing "None" as value here is not supported and leads to an exception - and i'm actually surprised mypy isn't screaming at you - as tye type-hint on `fetch_l2_order_book()` defines it as int - not `Optional[int]`.

I'm anyway not sure why you're using None as first argument (or why you're looping at all)